python/examples/helpers/io.py

```diff
@@ -24,27 +22,31 @@ def __iter__(self) -> "ConsoleReader":
         return self
 
     def __next__(self) -> str:
-        while True:
-            prompt = input(colored(self.input, "cyan", attrs=["bold"])).strip()
+        try:
+            while True:
+                prompt = input(colored(self.input, "cyan", attrs=["bold"])).strip()
 
-            if prompt == "q":
-                raise StopIteration
+                if prompt == "q":
+                    raise StopIteration
 
-            prompt = prompt if prompt else self.fallback
+                prompt = prompt if prompt else self.fallback
 
-            if not prompt and not self.allow_empty:
-                print("Error: Empty prompt is not allowed. Please try again.")
-                continue
+                if not prompt and not self.allow_empty:
+                    print("Error: Empty prompt is not allowed. Please try again.")
+                    continue
 
-            return prompt
+                return prompt
+        except (EOFError, KeyboardInterrupt):
+            print("Interactive session has ended.")
```

Comment:
I don't think this print statement is desired; otherwise, it is LGMT.

Reply:
I was adding a newline (so the next console prompt doesn't print on the same line) and though might as well add a message, but if you'd rather not I can remove it

Reply:
Please do so.

Reply:
done